tests/ssl-opt.sh

```diff
@@ -251,6 +251,24 @@ requires_any_configs_disabled() {
     fi
 }
 
+TLS1_2_KEY_EXCHANGES_WITH_CERT="MBEDTLS_KEY_EXCHANGE_RSA_ENABLED \
+                                MBEDTLS_KEY_EXCHANGE_DHE_RSA_ENABLED \
+                                MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED \
+                                MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED \
+                                MBEDTLS_KEY_EXCHANGE_RSA_PSK_ENABLED \
+                                MBEDTLS_KEY_EXCHANGE_ECDH_RSA_ENABLED \
+                                MBEDTLS_KEY_EXCHANGE_ECDH_ECDSA_ENABLED"
+
+requires_key_exchange_with_cert_in_tls12_or_tls13_enabled() {
```

Comment:
Is this supposed to be a corrected implementation of `requires_certificate_authentication`, or does it have different intended semantics?

I'm surprised that this pull requests adds a lot of explicit calls to `requires_key_exchange_with_cert_in_tls12_or_tls13_enabled`. We already have quite a bit of automatic detection of certificate requirements. Do you think these explicit calls would be needed if `requires_certificate_authentication` was fixed to handle TLS 1.3 and builds with PSK-ephemeral but no cert?